### PR TITLE
External links API

### PR DESCRIPTION
**Pull request by LachlanStuart**
Part of #507

This adds "external links" to datasets and projects for MetaboLights integration.

I also noticed that `Project.reviewToken` was unsecured, so I added a resolver to ensure only project managers can see it.

I did some cleaning up in python-client, added tests for the new functions and added the option to load configuration from a file.

### CODE REVIEW

File: metaspace/python-client/metaspace/tests/test_sm_instance.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+def test_add_dataset_external_link(sm, my_ds_id):
+    provider = 'MetaboLights'
+    link = 'https://www.ebi.ac.uk/metabolights/MTBLS313'
+
+    result = sm.add_dataset_external_link(my_ds_id, provider, link)
+
+    assert any(ext_link == {'provider': provider, 'link': link} for ext_link in result)
+
+
+def test_remove_project_external_link(sm, my_ds_id):
```
Comment (intsco) [inline metaspace/python-client/metaspace/tests/test_sm_instance.py]: `test_remove_project_external_link` -> `remove_dataset_external_link`

File: metaspace/graphql/src/modules/dataset/controller/Mutation.ts [hunk 2]
```diff
@@ -371,7 +373,37 @@ const MutationResolvers: FieldResolversFor<Mutation, void>  = {
...
+    source,
+    {datasetId, provider, link, replaceExisting},
+    ctx: Context
+  ) => {
+    await ctx.entityManager.transaction(async txn => {
+      const ds = await getDatasetForEditing(txn, ctx.user, datasetId);
+      await txn.update(DatasetModel, ds.id, {
+        externalLinks: addExternalLink(ds.externalLinks, provider, link, replaceExisting),
+      });
+    });
```
[the remark this answers is not in the document]
Comment (LachlanStuart) [reply metaspace/graphql/src/modules/dataset/controller/Mutation.ts]: Yeah. This is relatively low risk code, but it's such an obvious read-change-write pattern that I couldn't help but wrap it in a transaction.

File: metaspace/graphql/src/tests/graphqlTestEnvironment.ts [hunk 3]
```diff
@@ -40,6 +40,14 @@ const createTransactionEntityManager = async () => {
...
     try { await txn; } catch {}
   };
+  // Replace transaction implementation to prevent issues with nested transactions
+  manager.transaction = async (...args: any[]) => {
+    if (args.length == 1) {
+      return await args[0](manager);
+    } else {
+      return await args[1](manager);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (LachlanStuart) [reply metaspace/graphql/src/tests/graphqlTestEnvironment.ts]: Yes. TypeORM doesn't support nested transactions unfortunately. 

Tests wrap the `context.entityManager` in a global transaction (meaning any attempt to make a new transaction with that entityManager will fail), but when code is running normally there isn't a global transaction, so there's no problem.

File: metaspace/python-client/metaspace/__init__.py [hunk 4]
```diff
@@ -1,2 +1,2 @@
 name = 'metaspace2020'
-__version__ = '1.6.1'
+__version__ = '1.7.0'
```
[the remark this answers is not in the document]
Comment (LachlanStuart) [reply metaspace/python-client/metaspace/__init__.py]: I'd keep major/minor in sync with METASPACE, but change patch freely.

This PR adds new functionality, so by semver rules I'm bumping up the minor version.

File: metaspace/python-client/metaspace/tests/test_sm_instance.py (hunk 1 above)
Comment (LachlanStuart) [reply metaspace/python-client/metaspace/tests/test_sm_instance.py]: Fixed

File: metaspace/python-client/metaspace/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (intsco) [reply metaspace/python-client/metaspace/__init__.py]: I wonder if we should bump the M\  version to `1.7.0` too then.

**Pull request merged**